### PR TITLE
Rename StripeCustomer

### LINKED ISSUE #471: Rename StripeCustomer to StripePlatformCustomer
# Problem

We need to change the `StripeCustomer` to a `StripePlatformCustomer` that has many `StripeConnectCustomers`.

### PR DESCRIPTION
**Pull request by joshsmith**
# What's in this PR?

- [x] Update API Blueprint
- [x] Rename StripeCustomer to StripePlatformCustomer everywhere
- [x] Add StripeConnectCustomer

## References
Fixes #471

### CODE REVIEW

File: priv/repo/migrations/20161121014050_create_stripe_connect_customers.exs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+defmodule CodeCorps.Repo.Migrations.CreateStripeConnectCustomers do
+  use Ecto.Migration
+
+  def change do
+    create table(:stripe_connect_customers) do
+      add :id_from_stripe, :string, null: false
+
+      add :stripe_account_id, references(:stripe_accounts, on_delete: :nothing), null: false
+      add :stripe_platform_customer_id, references(:stripe_platform_customers, on_delete: :nothing), null: false
```
Comment (sbatson5) [inline priv/repo/migrations/20161121014050_create_stripe_connect_customers.exs:9]: Isn't `nothing` the default `on_delete` behavior?

File: test/policies/stripe_platform_customer_policy_test.exs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    end
+
+    test "returns true when user is viewing their own information" do
+      user = insert(:user)
+      stripe_platform_customer = insert(:stripe_platform_customer, user: user)
+
+      assert show?(user, stripe_platform_customer)
+    end
+
+    test "returns false when user id is not the StripePlatformCustomer's user_id" do
```
Comment (sbatson5) [inline test/policies/stripe_platform_customer_policy_test.exs:21]: 👏 nice test coverage

**Review by sbatson5 — APPROVED** [review]
Nothing stands out to me. Looks good! 👍

File: priv/repo/migrations/20161121014050_create_stripe_connect_customers.exs (hunk 1 above)
Comment (joshsmith) [reply priv/repo/migrations/20161121014050_create_stripe_connect_customers.exs:9]: I think so, but just wanted to be explicit here.

**Pull request merged**